packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -211,6 +217,41 @@ abstract class TextSelectionControls {
   }
 }
 
+/// Text selection controls that do not show any toolbars or handles.
+///
+/// The [emptyTextSelectionControls] global variable has a
+/// suitable instance of this class.
+class EmptyTextSelectionControls extends TextSelectionControls {
```

Comment:
Is this only used in test? The selection toolbar is also used as right click context menu. I am not sure if this can be useful to customer. should we make this private?

Reply:
It's useful in contexts that want to bootstrap and not adopt Material. I doubt it'd ship in a production app, except maybe in an embedded scenario, but it's still useful for developers.

I can add more docs cautioning about its limitations if you like though, what would you like it to say?

Reply:
this will likely to just be a placeholder until they wrote their own controls. To me, this feel as the same as we make _NullWidget public.

Reply:
that being said, we do make PlaceHolder widget public....

yeah, maybe just mention this is a placeholder and user will not be able to interact with the controls

Reply:
done